packages/python/src/armonik/client/events.py

```diff
@@ -0,0 +1,63 @@
+from ..protogen.common.events_common_pb2 import EventSubscriptionRequest, EventSubscriptionResponse
+from ..protogen.common.results_filters_pb2 import Filters as rawResultFilters
+from ..protogen.common.tasks_filters_pb2 import Filters as rawTaskFilters
+
+class ArmoniKEvents:
+
+    _events_obj_mapping = {
+        "new_result": NewResultEvent,
+        "new_task": NewTaskEvent,
+        "result_owner_update": ResultOwnerUpdateEvent,
+        "result_status_update": ResultStatusUpdateEvent,
+        "task_status_update": TaskStatusUpdateEvent
+    }
+
+    def __init__(self, grpc_channel: Channel):
+        """Events service client
+
+        Args:
+            grpc_channel: gRPC channel to use
+        """
+        self._client = EventsStub(grpc_channel)
+
+    def get_events(self, session_id: str, event_types: List[EventTypes], event_handlers: List[Callable[[str, Any], bool]], task_filter: Filter, result_filter: Filter) -> None:
+        """Get events that represents updates of result and tasks data.
+        
+        Args:
+            session_id: The ID of the session.
+            event_types: The list of the types of event to catch.
+            event_handlers: The list of handlers that process the events. Handlers are evaluated in he order they are provided.
+                An handler takes three positional arguments: the ID of the session, the type of event and the event as an object.
+                An handler returns a boolean, if True the process continues, otherwise the stream is closed and the service stops
+                listening to new events.
+            task_filter: A filter on tasks.
+            result_filter: A filter on results.
+
+        """
+        request = EventSubscriptionRequest(
+            session_id=session_id,
+            tasks_filters=cast(rawTaskFilters, task_filter.to_disjunction().to_message()),
+            results_filters=cast(rawResultFilters, result_filter.to_disjunction().to_message()),
+            returned_events=event_types
+        )
+        streaming_call = self._client.GetEvents(request)
+        for message in streaming_call:
+            event_type = message.WhichOneof("update")
+            if any([event_handler(session_id, EventTypes.from_string(event_type), self._events_obj_mapping[event_type].from_raw_event(getattr(message, event_type))) for event_handler in event_handlers]):
+                break
+
+    def wait_for_result_availability(self, result_id: str, session_id: str) -> bool:
+        def handler(session_id, event_type, event):
+            if not isinstance(event, ResultStatusUpdateEvent):
+                raise ValueError("Handler should receive event of type 'ResultStatusUpdateEvent'.")
+            if event.status == ResultStatus.COMPLETED:
+                return False
+            return True
+        self.get_events(session_id, [EventTypes.RESULT_STATUS_UPDATE], lambda _: False, result_filter=(ResultFieldFilter.RESULT_ID == result_id))
```

Comment:
you should also consider results creations. If you subscribe after the results are completed, you will receive only one result creation event with status created.